rgw: send headers of quota settings

ofriedma opened the pull request:
Make the quota settings visible to s3 and swift users
rgw send headers of quota settings

Fixes: https://tracker.ceph.com/issues/47752
s3-tests: https://github.com/ceph/s3-tests/pull/371
Signed-off-by: Or Friedmann <ofriedma@redhat.com>

mattbenjamin commented on src/rgw/rgw_rest_s3.cc:
```diff
@@ -2140,6 +2140,14 @@ static void dump_bucket_metadata(struct req_state *s, rgw::sal::RGWBucket* bucke
+  // only bucket's owner is allowed to get the quota settings of the account
+  if (bucket->is_owner(s->user.get())) {
+    auto user_info = s->user->get_info();
+    dump_header(s, "X-RGW-Quota-Bucket-size", static_cast<long long>(user_info.bucket_quota.max_size));
```
capitalization doesn't match the existing headers

ofriedma commented:
need to change it to capital letter

ofriedma commented:
jenkins test make check

ofriedma commented:
jenkins test docs

ofriedma commented:
jenkins test make check

ofriedma commented:
I Added: 
1) quota info to getusage api and fixed it (it caused segfault in master):
https://docs.ceph.com/en/latest/radosgw/s3/serviceops/#get-usage-stats
2) max buckets header to head bucket op
3) Added head bucket header printing to boto3 examples
@dang @mattbenjamin Can you please take a look again?

ofriedma commented:
http://pulpito.ceph.com/ofriedma-2020-10-19_19:14:20-rgw-wip-ofriedma-show-quota-stats-i01-distro-basic-smithi/5539497/

ofriedma commented:
@dang @mattbenjamin Do you think we can merge it?

mattbenjamin reviewed (approved):
lgtm; looks like teuthology run is a pass
